weblogic-azure-aks/src/main/arm/scripts/applyGuaranteedQos.sh

```diff
@@ -0,0 +1,137 @@
+# Copyright (c) 2021, Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+# This script runs on Azure Container Instance with Alpine Linux that Azure Deployment script creates.
+#
+# env inputs:
+# AKS_CLUSTER_NAME
+# AKS_CLUSTER_RESOURCEGROUP_NAME
+# WLS_DOMAIN_UID
+
+# Main script
+script="${BASH_SOURCE[0]}"
+scriptDir="$(cd "$(dirname "${script}")" && pwd)"
+source ${scriptDir}/common.sh
+source ${scriptDir}/utility.sh
+
+qualityofService="BestEffort"
+wlsContainerName="weblogic-server"
+wlsDomainNS="${WLS_DOMAIN_UID}-ns"
+
+echo_stdout "install kubectl"
+install_kubectl
+
+echo_stdout "Connect AKS"
+az aks get-credentials \
+    --resource-group ${AKS_CLUSTER_RESOURCEGROUP_NAME} \
+    --name ${AKS_CLUSTER_NAME} \
+    --overwrite-existing
+
+adminPodName=$(kubectl -n ${wlsDomainNS} get pod -l weblogic.serverName=admin-server -o json |
+    jq '.items[0] | .metadata.name' |
+    tr -d "\"")
+if [ -z "${adminPodName}" ]; then
+    echo_stderr "Fail to get admin server pod."
+    exit 1
+fi
+
+wlstQueryVersionScript=queryVersion.py
+cat <<EOF >${wlstQueryVersionScript}
+print '#version#:' + version
+EOF
+
+echo_stdout "copy WLST script ${wlstQueryVersionScript} to ${adminPodName}:/tmp/${wlstQueryVersionScript}"
+targetPyFilePath=/tmp/${wlstQueryVersionScript}
+kubectl cp ${wlstQueryVersionScript} -n ${wlsDomainNS} ${adminPodName}:${targetPyFilePath}
+version=$(kubectl exec -it ${adminPodName} -n ${wlsDomainNS} -c ${wlsContainerName} -- bash -c "wlst.sh ${targetPyFilePath}")
+# output sample:
+# Initializing WebLogic Scripting Tool (WLST) ...
+
+# Welcome to WebLogic Server Administration Scripting Shell
+
+# Type help() for help on available commands
+
+# #version#:WebLogic Server 14.1.1.0.0
+version="${version##*\#version\#\:}" # match #version#:, this is a special mark for the version output, please do not change it.
+echo_stdout ${version}
+
+if [ "${version#*WebLogic Server 14.1.1.0}" != "$version" ]; then
```

Comment:
I have not tried to fully understand this script so I could be wrong, but even so, I have two comments:
- To future proof, assuming the problem could be in all future 14.x versions, should the version check be a "greater than or equal"?  I think you can leverage `sort --version-sort` for this.
- Would the check be better off running before deploying the domain resource? E.g. a `kubectl run` equivalent of `docker run SOME_IMAGE_NAME sh -c 'source $ORACLE_HOME/wlserver/server/bin/setWLSEnv.sh > /dev/null 2>&1 && java weblogic.version'`.

Reply:
Thanks for your review @tbarnes-us  Wish you a Merry Christmas though it's a late greeting.

For #1, you are right, we should consider future version, while I would like to only apply the workaround to 14.1.1.0 now, as we con not make sure the issue will happen in the future version, and we may have other configuration to resolve the issue after we find out the root cause.  Also cc @edburns 

For #2, we can check the version before the deploying the domain resource. Considering the usage of post deployment for data source set up, we separate the processes of basic cluster set up and other configuration that configured via configmap which requires a restart.   Current work flow is:

- Deploy the basic without data source
- [Optional if the customer select data source] deploy configmap for data source and patch the cluster, this will cause a restart
- Apply resource limits to WLS 14c.

The domain deployment will be much slower by applying resource limits (2min/pod to 8min/pod). So we apply the setting after there is not further restart(rolling update) process.


For getting version using `docker run`, this is a great idea, without executing the script using `kubectl exec`.
We have to run the command in a machine with docker deamon, I will check and add another commit.

Reply:
For #2 I was thinking of `kubectl run`. This implicitly launches a temporary pod in the k8s cluster, and so has the advantage of using the same docker image cache as the k8s cluster. One issue is that it may need a pull secret - not sure how that'd be done.

Reply:
For #2:> while I would like to only apply the workaround to 14.1.1.0 now

Reply:
> For #2:
> 
> > while I would like to only apply the workaround to 14.1.1.0 now
> 
> That's fine by me - as long as one of us remembers in the case that we still need the work-around once 14.1.2 comes out.

We can create an issue to track that, let's discuss with Ed when he is back next week.

Reply:
> For #2 I was thinking of `kubectl run`. This implicitly launches a temporary pod in the k8s cluster, and so has the advantage of using the same docker image cache as the k8s cluster. One issue is that it may need a pull secret - not sure how that'd be done.

We have the domain running when running the shell script to apply resources limits, I enhanced the script using `java weblogic.version`, please find https://github.com/oracle/weblogic-azure/pull/103/commits/55233475d88f3c69d71c09e7bb07a79c49c8fc0b